packages/flutter/test/cupertino/date_picker_test.dart

```diff
@@ -2414,6 +2415,41 @@ void main() {
 
     expect(find.byType(CupertinoPickerDefaultSelectionOverlay), findsExactly(4));
   });
+
+  testWidgets('CupertinoDatePicker getColumnWidth', (WidgetTester tester) async {
+    const List<String> testWords = <String>[
```

Comment:
What is this test verifying? I thought for this bug fix we want to verify that `CupertinoDatePicker.getColumnWidth` works correctly when text with fewer characters is actually longer than text with more characters?

Reply:
The test should fail without the patch. But it doesn't seem to be the case?

Reply:
I have some doubts about this. I tried to check if the difference is caused by the automatically applied kerning, but I didn't find anything specific. After adding the Roboto font, I got the following results:
|Word   | Roboto (width) | CupertinoSystemDisplay (width) |
|-----------|------------------|----------------------------------|
| February  | 85.0             | 171.0                            |
| November  | 100.0            | 171.0                            |
| Fevereiro | 89.0             | 193.0                            |
| Novembro  | 100.0            | 171.0                            |



### LineMetrics 

| Field               | CupertinoSystemDisplay                        | Roboto                             |
|---------------------|--------------------------------------|---------------------------------------|
| **hardBreak**       | `true`                                | `true`                                 |
| **ascent**          | `15.75`                               | `19.482421875`                         |
| **descent**         | `5.25`                                | `5.126953125`                          |
| **unscaledAscent**  | `15.75`                               | `19.482421875`                         |
| **height**          | `21.0`                                | `25.0`                                 |
| **width**           | `85.5999984741211`                    | `36.852928161621094`                   |
| **left**            | `0.20000000298023224`                 | `0.20000000298023224`                  |
| **baseline**        | `15.75`                               | `19.873046875`                         |
| **lineNumber**      | `0`                                   | `0`                                    |

Reply:
In tests typically the test fonts are used, not the real fonts.

Reply:
I wouldn't recommend loading fonts from `flutterRoot.childFile(fs.path.join('bin', 'cache', 'artifacts', 'material_fonts', fontPath));`. The default test font has some characters with smaller x-advances. You can try those called "1/x Advance` glyphs.

Reply:
I believe I have now achieved the result I was hoping for.